test/demo_client.py

```diff
@@ -1,32 +1,11 @@
 import os
-import requests
-from requests.adapters import HTTPAdapter
 import logging
 
 logging.basicConfig(
     level=logging.INFO, 
     format='[%(asctime)s] {%(module)s:%(lineno)d} %(levelname)s - %(message)s'
 )
 
-# create http adapter with exponential backoff (for unstable and/or slow connections)
-http_adapter = HTTPAdapter(
-    max_retries=requests.packages.urllib3.util.retry.Retry(
-        total=5,
-        backoff_factor=10,
-        status_forcelist=[429,500,502,503,504]
-    )
-)
-custom_session = requests.session()
-custom_session.mount('http://', http_adapter)
-custom_session.mount('https://', http_adapter)
-
-# use os env proxy settings, if any
-custom_session.proxies.update({
-    'http' : os.environ.get('http_proxy',''),
-    'https' : os.environ.get('http_proxy', '')
-})
-
-
```

Comment:
Only Python experts with deep understanding of urllib3 will create a custom session. I believe it is better that we create one with sensible defaults upon Client initialization. This will serve the purposes of most users.